ci/download-workflow-artifacts.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+set -e
+
+# This script downloads and unzips the artifacts produced in a workflow run. It
+# also checks that the workflow commit corresponds to the tag commit that these
+# artifacts will be released under. The script has several pre-requisites:
+# - some standard Bash tools (curl, unzip) and one slightly more rare one (jq)
+# - an already-created tag in the repository (this marks the code to release)
+# - the ID of a workflow run that has run successfully--this is where we
+#   retrieve the artifacts from
+# - a GitHub access token, see https://github.com/settings/tokens
+#
+# Usage: download-workflow-artifacts.sh <release tag> <workflow run ID> <token>
+
+TAG=$1
+WORKFLOW_RUN_ID=$2
+GITHUB_TOKEN=$3
+GITHUB_API_VERSION=2022-11-28
+GITHUB_API_URL=https://api.github.com/repos/WebAssembly/wasi-sdk
```

Comment:
why do you prefer to have a low-level script here while there are many handy commands/scripts (eg. `hub`) and even github actions available?

Reply:
Not everyone has all of that installed; I went with an environment I thought @sunfishcode would like 😁. (The idea was that these scripts would be runnable both from CI or from a maintainer's shell if something goes wrong).

Reply:
i suspect it's easier to install one of them than maintaining this script by ourselves. but i don't insist.